src/main/java/com/github/webicitybrowser/spec/http/HTTPRequest.java

```diff
@@ -1,7 +1,12 @@
 package com.github.webicitybrowser.spec.http;
 
 import com.github.webicitybrowser.spec.url.URL;
+import com.github.webicitybrowser.webicity.core.net.ProtocolContext;
 
 public record HTTPRequest(URL url, String method, HTTPRedirectHandler redirectHandler) {
 
+	public static HTTPRequest createRequest(URL url, ProtocolContext context) {

```

Comment:
This is a no-go.  Things in the `spec` package should not depend on things on the `webicity` package. You can think of them as separate modules, almost, with a one-way dependency.